frontend/src/lib/components/elements/Toast/Toast.tsx

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  ReactElement,
+  useState,
+  useEffect,
+  useCallback,
+  useMemo,
+} from "react"
+import { withTheme } from "@emotion/react"
+import { toaster, ToastOverrides } from "baseui/toast"
+
+import { hasLightBackgroundColor, EmotionTheme } from "src/lib/theme"
+import StreamlitMarkdown from "src/lib/components/shared/StreamlitMarkdown"
+
+import { StyledViewButton, StyledToastMessage } from "./styled-components"
+import { Kind } from "src/lib/components/shared/AlertContainer"
+import AlertElement from "src/lib/components/elements/AlertElement/AlertElement"
+
+export interface ToastProps {
+  theme: EmotionTheme
+  text: string
+  icon?: string
+  width: number
+}
+
+function generateToastOverrides(
+  expanded: boolean,
+  theme: EmotionTheme
+): ToastOverrides {
+  const lightBackground = hasLightBackgroundColor(theme)
+  return {
+    Body: {
+      props: {
+        "data-testid": "stToast",
+      },
+      style: {
+        width: "288px",
+        marginTop: "8px",
+        borderRadius: "4px",
```

Comment:
nit: would be nice to move these into the theme object if there are reasonable places to put them (same comment for a few other magic values in this file + throughout the PR)